server/src/main/java/org/opensearch/search/optimization/filterrewrite/DateHistogramAggregatorBridge.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    private Ranges buildRanges(long[] bounds) {
+        bounds = processHardBounds(bounds);
+        if (bounds == null) {
+            return null;
+        }
+        assert bounds[0] <= bounds[1] : "Low bound should be less than high bound";
+
+        final Rounding rounding = getRounding(bounds[0], bounds[1]);
+        final OptionalLong intervalOpt = Rounding.getInterval(rounding);
+        if (intervalOpt.isEmpty()) {
+            return null;
+        }
+        final long interval = intervalOpt.getAsLong();
+
+        // process the after key of composite agg
+        bounds = processAfterKey(bounds, interval);
+
+        return Helper.createRangesFromAgg(
+            (DateFieldMapper.DateFieldType) fieldType,
+            interval,
+            getRoundingPrepared(),
+            bounds[0],
+            bounds[1],
+            optimizationContext.maxAggRewriteFilters
+        );
+    }
+
+    protected abstract Rounding getRounding(final long low, final long high);
+
+    protected abstract Rounding.Prepared getRoundingPrepared();
+
+    protected long[] processAfterKey(long[] bounds, long interval) {
+        return bounds;
+    }
+
+    protected long[] processHardBounds(long[] bounds) {
+        return processHardBounds(bounds, null);
+    }
+
+    protected long[] processHardBounds(long[] bounds, LongBounds hardBounds) {
+        if (bounds != null) {
+            // Update min/max limit if user specified any hard bounds
+            if (hardBounds != null) {
+                if (hardBounds.getMin() > bounds[0]) {
+                    bounds[0] = hardBounds.getMin();
+                }
+                if (hardBounds.getMax() - 1 < bounds[1]) {
+                    bounds[1] = hardBounds.getMax() - 1; // hard bounds max is exclusive
+                }
+                if (bounds[0] > bounds[1]) {
+                    return null;
+                }
+            }
+        }
+        return bounds;
+    }
+
+    private DateFieldMapper.DateFieldType getFieldType() {
+        assert fieldType instanceof DateFieldMapper.DateFieldType;
+        return (DateFieldMapper.DateFieldType) fieldType;
+    }
+
+    protected int getSize() {
+        return Integer.MAX_VALUE;
+    }
+
+    @Override
+    public final void tryOptimize(PointValues values, BiConsumer<Long, Long> incrementDocCount) throws IOException {
+        int size = getSize();
+
+        DateFieldMapper.DateFieldType fieldType = getFieldType();
+        BiConsumer<Integer, Integer> incrementFunc = (activeIndex, docCount) -> {
+            long rangeStart = LongPoint.decodeDimension(optimizationContext.getRanges().lowers[activeIndex], 0);
+            rangeStart = fieldType.convertNanosToMillis(rangeStart);
+            long ord = getBucketOrd(bucketOrdProducer().apply(rangeStart));
+            incrementDocCount.accept(ord, (long) docCount);
+        };
+
+        optimizationContext.consumeDebugInfo(
```

Comment:
you can remove this component's dependency on optimizationContext by having `tryOptimize` return `OptimizationContext.DebugInfo`

Reply:
Updated as this method is actually pkg-private